packages/model-viewer/src/constants.ts

```diff
@@ -70,6 +68,8 @@ export const IS_AR_QUICKLOOK_CANDIDATE = (() => {
       tempAnchor.relList.supports('ar'));
 })();
 
+export const IS_WEBXR_AR_CANDIDATE = IS_ANDROID && HAS_WEBXR_HIT_TEST_API;
```

Comment:
Principally speaking, it is an anti-pattern to base the notion of web platform feature support on characteristics such as device or brother version. We do it where we have to when there are corners of the world that are actually proprietary to a device, browser or manufacturer. But, in general we should avoid qualifications like this one unless absolutely necessary.

If there is a case where a device partially (or erroneously) implements a spec, then I recommend doing feature detection for that particular case instead.

Reply:
Agreed, though this is not erroneously implemented; it's just that the presence of the hit test API does not actually say anything about AR being supported or not. The proper way to test this is `await navigator.xr.isSessionSupported("immersive-ar")`, which is async because it will try to download ARCore if it's not already on the device (and there is some associated system UI). Right now we only attempt this when someone tries to enter AR. However, our constant is there so we can choose whether to set the renderer to preserve its drawing buffer, which we have to decide when we create it. 

Should we move the possible ARCore download to page load in order to get more accurate knowledge? Or just set preserve draw buffer all the time and not worry about any performance effect that might have? I think all three options are valid enough for discussion.

Reply:
Hey, nevermind! Turns out the preservedrawingbuffer is vestigial anyway, so I can remove this whole code path. It was for an old version of WebXR.